File: app-android/app/src/main/java/com/health2u/data/local/entity/EmergencyContactEntity.kt
package com.health2u.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.Index

@Entity(
    tableName = "emergency_contacts",
    indices = [Index(value = ["userId"]), Index(value = ["order"])]
)
data class EmergencyContactEntity(
    @PrimaryKey val id: String,
    val userId: String,
    val name: String,
    val relationship: String,
    val phone: String,
    val email: String?,
    val isPrimary: Boolean,
    val order: Int
)

File: app-ios/Sources/Health2u/Data/Local/EmergencyContactEntity.swift
import Foundation

public struct EmergencyContactEntity: Sendable, Equatable, Identifiable {
    public let id: String
    public var userId: String
    public var name: String
    public var relationship: String
    public var phone: String
    public var email: String?
    public var isPrimary: Bool
    public var order: Int

    public init(
        id: String,
        userId: String,
        name: String,
        relationship: String,
        phone: String,
        email: String? = nil,
        isPrimary: Bool,
        order: Int
    ) {
        self.id = id
        self.userId = userId
        self.name = name
        self.relationship = relationship
        self.phone = phone
        self.email = email
        self.isPrimary = isPrimary
        self.order = order
    }
}
